File: CS/11_SecurityAndSignatures/Encryption.cs
using System;
using System.Drawing;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;
using Spire.Pdf.Security;

namespace Encryption
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {        
            // Create and load a pdf document
            PdfDocument pdf = new PdfDocument();
            pdf.LoadFromFile(@"..\..\..\..\..\..\Data\Encryption.pdf");

            // Define user and owner passwords
            string userPassword = "user";
            string ownerPassword = "owner";

            // Create a security policy with the specified passwords
            PdfSecurityPolicy securityPolicy = new PdfPasswordSecurityPolicy(userPassword, ownerPassword);

            // Set the encryption algorithm to AES 128-bit
            securityPolicy.EncryptionAlgorithm = PdfEncryptionAlgorithm.AES_128;

            // Allow printing of the document
            securityPolicy.DocumentPrivilege.AllowPrint = true;

            // Allow filling form fields in the document
            securityPolicy.DocumentPrivilege.AllowFillFormFields = true;

            // Allow copying content from the document
            securityPolicy.DocumentPrivilege.AllowContentCopying = true;

            // Encrypt the PDF document using the specified security policy
            pdf.Encrypt(securityPolicy);

            // Save the encrypted PDF document to a file named "SecurityPermission.pdf"
            pdf.SaveToFile("Encryption-result.pdf");
          
            //Launch the file.
            PDFDocumentViewer("Encryption-result.pdf");
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}

File: VB/11_SecurityAndSignatures/Encryption.vb
Imports Spire.Pdf

Namespace Encryption
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Load a PDF document
			Dim doc As New PdfDocument()
			doc.LoadFromFile("..\..\..\..\..\..\Data\Encryption.pdf")

			' Specify the open password and permission password for the encryption
			Dim openPassword As String = "e-iceblue"
			Dim permissionPassword As String = "test"

			' Create a new PdfPasswordSecurityPolicy object with the specified passwords
			Dim policy As New PdfPasswordSecurityPolicy(openPassword, permissionPassword)

			' Set the encryption algorithm to AES-256
			policy.EncryptionAlgorithm = PdfEncryptionAlgorithm.AES_128

			' Allow printing and filling form fields in the document
			policy.DocumentPrivilege.AllowPrint = True
			policy.DocumentPrivilege.AllowFillFormFields = True

			' Encrypt the PDF document using the specified security policy
			doc.Encrypt(policy)

			' Save the encrypted PDF document to a file
			doc.SaveToFile("Encryption-result.pdf")

			' Close the document
			doc.Close()

			' Launch the file
			PDFDocumentViewer("Encryption-result.pdf")
		End Sub

		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace
